src/mixins/canvas_events.mixin.js

```diff
@@ -207,40 +212,179 @@
       this.__onLongPress && this.__onLongPress(e, self);
     },
 
+    /**
+     * supports native like text dragging
+     * @private
+     * @param {DragEvent} e
+     */
+    _onDragStart: function (e) {
+      var activeObject = this.getActiveObject();
+      if (activeObject && typeof activeObject.onDragStart === 'function' && activeObject.onDragStart(e)) {
+        this._dragSource = activeObject;
+        var options = { e: e, target: activeObject };
+        this.fire('dragstart', options);
+        activeObject.fire('dragstart', options);
+        addListener(this.upperCanvasEl, 'drag', this._onDragProgress);
+        return;
+      }
+      e.preventDefault();
+      e.stopPropagation();
+    },
```

Comment:
i would prefer to call this `_onDragStartTextInternal` ( or similar ) and short circuit if the active object type does not inherit from text rather than checking if the activeObject has onDragStart function.

If more drag/drop functionality get added in the future, each functionality can have its own handler and they can be turned off easily by not registering the event handler or de-registering it.

Reply:
I disagree because this way any object can utilize this logic out of the box.
And the place to disable the event is from `onDragStart`  - this is from the spec.

Reply:
I don't want to open the doors to drag things that we didn't plan to support.
We are not adding text drag support, with a skeleton of infrastructure to do maybe more dragging, but we don't really know.
It is easier to add later than rollback